apps/root/src/pages/position-detail/components/summary-container/components/position-data/index.tsx

```diff
@@ -240,36 +85,41 @@ const Details = ({ position, pendingTransaction, yieldOptions, totalGasSaved }:
     tokenToAverage.address === PROTOCOL_TOKEN_ADDRESS
       ? { ...wrappedProtocolToken, symbol: tokenToAverage.symbol, underlyingTokens: tokenToAverage.underlyingTokens }
       : tokenToAverage;
-  swappedActions.forEach((action) => {
+
+  const summedPrices = swappedActions.reduce((acc, action) => {
     const swappedRate =
       action.tokenA.address ===
       ((tokenFromAverage.underlyingTokens[0] && tokenFromAverage.underlyingTokens[0].address) ||
         tokenFromAverage.address)
-        ? BigInt(action.ratioAToB)
-        : BigInt(action.ratioBToA);
+        ? action.ratioBToA
+        : action.ratioAToB;
+    return acc + swappedRate;
+  }, 0n);
 
-    summedPrices = summedPrices + swappedRate;
-  });
   const intl = useIntl();
+
   const averageBuyPrice = summedPrices > 0n ? summedPrices / BigInt(swappedActions.length) : 0n;
 
+  const totalDeposited = position.history?.reduce<bigint>((acc, event) => {
+    let newAcc = acc;
+    if (event.action === ActionTypeAction.CREATED) {
+      newAcc += event.rate * BigInt(event.swaps);
+    } else if (event.action === ActionTypeAction.MODIFIED) {
+      newAcc += event.rate * BigInt(event.remainingSwaps) - event.oldRate * BigInt(event.oldRemainingSwaps);
+    }
+    return newAcc;
+  }, 0n);
+
   const fromUsdPrice = prices?.fromPrice;
   const toUsdPrice = prices?.toPrice;
+  const showFromPrice = !!fromUsdPrice;
+  const showToPrice = !!toUsdPrice;
 
-  const fromPrice = parseUsdPrice(position.from, BigInt(remainingLiquidity), fromUsdPrice);
-  const fromYieldPrice = parseUsdPrice(position.from, BigInt(yieldFromGenerated?.amount || 0n), fromUsdPrice);
   const ratePrice = parseUsdPrice(position.from, rate.amount, fromUsdPrice);
-  const toPrice = parseUsdPrice(position.to, toWithdrawBase, toUsdPrice);
-  const toYieldPrice = parseUsdPrice(position.to, toWithdrawYield?.amount, toUsdPrice);
-  const toFullPrice = parseUsdPrice(position.to, swappedBase, toUsdPrice);
-  const toYieldFullPrice = parseUsdPrice(position.to, swappedYield?.amount, toUsdPrice);
-  const showToFullPrice = !!toFullPrice;
-  const showToYieldFullPrice = !!toYieldFullPrice;
-  const showToPrice = !!toPrice;
-  const showToYieldPrice = !!toYieldPrice;
-  const showRatePrice = !!ratePrice;
-  const showFromPrice = !!fromPrice;
-  const showFromYieldPrice = !!fromYieldPrice;
+  const toWithdrawPrice = parseUsdPrice(position.to, toWithdraw.amount, toUsdPrice);
```

Comment:
The prices should actually already be inside the token :)

Reply:
Oh nice, forgot about this one!